File: src/main/java/retrospector/core/interactor/crud/CrudRequestRouter.java
package retrospector.core.interactor.crud;

import retrospector.core.boundry.Request;
import retrospector.core.boundry.RequestRouter;

public class CrudRequestRouter implements RequestRouter{
    private CrudMediaUseCase mediaUse;
    private CrudReviewUseCase reviewUse;
    private CrudFactoidUseCase factoidUse;
    private CrudRequest request;

    public CrudRequestRouter(CrudMediaUseCase mediaUse, CrudReviewUseCase reviewUse, CrudFactoidUseCase factoidUse) {
        this.mediaUse = mediaUse;
        this.reviewUse = reviewUse;
        this.factoidUse = factoidUse;
    }
    
    @Override
    public void disseminate(Request request) {
        this.request = (CrudRequest) request;
        
        if (request instanceof CrudMediaRequest)
            mediaUse.execute(request);
        else if (request instanceof CrudReviewRequest)
            reviewUse.execute(request);
        else if (request instanceof CrudFactoidRequest)
            factoidUse.execute(request);
    }
    
}

File: Retrospector.Core/Crud/CrudRequestRouter.cs
using Retrospector.Core.Boundary;
using Retrospector.Core.Crud.Models;

namespace Retrospector.Core.Crud
{
    public class CrudRequestRouter : IRequestRouter
    {
        private readonly IUseCase<CrudRequest<Media>> _mediaUse;
        private readonly IUseCase<CrudRequest<Review>> _reviewUse;
        private readonly IUseCase<CrudRequest<Factoid>> _factoidUse;

        public CrudRequestRouter(IUseCase<CrudRequest<Media>> mediaUse, IUseCase<CrudRequest<Review>> reviewUse, IUseCase<CrudRequest<Factoid>> factoidUse)
        {
            _mediaUse = mediaUse;
            _reviewUse = reviewUse;
            _factoidUse = factoidUse;
        }

        public void Disseminate(IRequest request)
        {
            switch (request)
            {
                case CrudRequest<Media> mediaRequest:
                    _mediaUse.Execute(mediaRequest);
                    break;
                case CrudRequest<Review> reviewRequest:
                    _reviewUse.Execute(reviewRequest);
                    break;
                case CrudRequest<Factoid> factoidRequest:
                    _factoidUse.Execute(factoidRequest);
                    break;
            }
        }
    }
}